runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10570,6 +10570,310 @@ inlineCompareAndSwapNative(
    }
 
 
+/**
+ * \brief
+ *   Generate inlined instructions equivalent to java/lang/StringCoding.hasNegatives or java/lang/StringCoding.countPositives
+ *
+ * \param node
+ *   The tree node
+ *
+ * \param cg
+ *   The Code Generator
+ */
+static TR::Register* inlineHasNegativesOrCountPositives(TR::Node* node, TR::RecognizedMethod recognizedMethod, TR::CodeGenerator* cg)
+   {
+   // Arguments to hasNegatives or countPositives
+   // Byte array
+   TR::Register *bufReg = cg->evaluate(node->getChild(0));
+   // Offset (i.e. index to begin counting from)
+   TR::Register *offsetReg = cg->evaluate(node->getChild(1));
+   // Length of byte array
+   TR::Register *lengthReg = cg->evaluate(node->getChild(2));
+
+   TR::Register *loopLimitReg = cg->allocateRegister();
+   TR::Register *limitReg = cg->allocateRegister();
+   TR::Register *maskReg = cg->allocateRegister();
+   TR::Register *indexReg = cg->allocateRegister();
+   TR::Register *xmmChunkReg = cg->allocateRegister(TR_VRF);
```

Comment:
Allocation of the `xmmChunkReg` should be guarded by the test of `pmovmskbEncoding`.  No need to allocate that register and add pre- and post-conditions for it in the case where it's not needed.